SimpleInventoryManagementSystem/InputLogger.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using SimpleInventoryManagementSystem;
+
+
+namespace SimpleInventoryManagementSystem;
+
+public class InputLogger
+{
+    ProductFactory _productFactory;
+
+    public InputLogger()
+    {
+        _productFactory = new ProductFactory();
+    }
+
+    public Product ProductInput()
+    {
+        var input = string.Empty;
+
+        Console.WriteLine("Adding new product to the inventory");
+
+        Console.WriteLine("Enter name of the product");
+        input = Console.ReadLine();
+        if(input is null)
+        {
+            throw new ArgumentException("invalid input");
+        }
+        var name = input;
+
+        Console.WriteLine("Enter price of the product");
+        input = Console.ReadLine();
+        if(!float.TryParse(input, out float price))
+        {
+            throw new ArgumentException("invalid input");
+        }
```

Comment:
`float.TryParse` already tells us which type is being parsed into. Let's start getting used to `var`.

```suggestion
        if(!float.TryParse(input, out var price))
        {
            throw new ArgumentException("invalid input");
        }
```